packages/tracing/src/browser/metrics.ts

```diff
@@ -0,0 +1,279 @@
+import { getGlobalObject, logger } from '@sentry/utils';
+
+import { Span } from '../span';
+import { Transaction } from '../transaction';
+
+import { msToSec } from './utils';
+
+const global = getGlobalObject<Window>();
+
+/** Class tracking metrics  */
+export class MetricsInstrumentation {
+  private _lcp: Record<string, any> = {};
+
+  private _performanceCursor: number = 0;
+
+  private _forceLCP = () => {
+    /* No-op, replaced later if LCP API is available. */
+    return;
+  };
+
+  /** Starts tracking the Largest Contentful Paint on the current page. */
+  private _trackLCP(): void {
+    // Based on reference implementation from https://web.dev/lcp/#measure-lcp-in-javascript.
+    // Use a try/catch instead of feature detecting `largest-contentful-paint`
+    // support, since some browsers throw when using the new `type` option.
+    // https://bugs.webkit.org/show_bug.cgi?id=209216
+    try {
+      // Keep track of whether (and when) the page was first hidden, see:
+      // https://github.com/w3c/page-visibility/issues/29
+      // NOTE: ideally this check would be performed in the document <head>
+      // to avoid cases where the visibility state changes before this code runs.
+      let firstHiddenTime = document.visibilityState === 'hidden' ? 0 : Infinity;
+      document.addEventListener(
+        'visibilitychange',
+        event => {
+          firstHiddenTime = Math.min(firstHiddenTime, event.timeStamp);
```

Comment:
Is it really `timeStamp` with capital `S`? :< sad panda

Reply:
Ya it kinda sucks lol https://developer.mozilla.org/en-US/docs/Web/API/Event/timeStamp